pkg/controller/spinnakervalidating/spinnakervalidating_controller.go

```diff
@@ -90,6 +95,26 @@ func (v *spinnakerValidatingController) Handle(ctx context.Context, req admissio
 	return admission.ValidationResponse(true, "")
 }
 
+func (v *spinnakerValidatingController) warnInValidationDelay(d time.Duration, svc v1alpha2.SpinnakerServiceInterface) chan bool {
+	ch := make(chan bool, 1)
+	go func() {
+		select {
+		case <-ch:
+			return
+		case <-time.After(d):
+			validations := halyard.GetValidationKeys()
+			msg := fmt.Sprintf("WARNING: Validations are taking too much time. Consider disabling individual validations in: "+
```

Comment:
To be even more useful, can we output the name of the validations that are still running? This would probably need to be moved under the `validate` probably under `parallel.go` but that would be for the best:
- controller code would remain lean and clean
- we know what validations are still running
- we already process things in parallel so it'd integrate nicely should we want to have the options to cancel pending validations in the future.

Reply:
Given that the process for creating the dynamic list of validators to run can also face some delays (like decrypting any remote secret), I think it's better to wrap the timeout around the full validations call in the controller, but we can still have the logic of which validations are currently running in `validate`.

Reply:
After some inspection of parallel validator, outputting the running validator names will need a bit more code changes than I would like for a patch release.